lib/rucio/cli/upload.py

```diff
@@ -35,8 +35,9 @@
 @click.option("--lfn", help="Specify the exact LFN for the upload")
 @click.option("--transfer-timeout", type=float, default=config_get_float("upload", "transfer_timeout", False, 360), help="Transfer timeout (in seconds)")
 @click.option("-r", "--recursive", is_flag=True, default=False, help="Convert recursively the folder structure into collections")
+@click.option("--new-file", is_flag=True, default=False, help="Assume DID does not already exist")
 @click.pass_context
-def upload_command(ctx, file_paths, rse, lifetime, expiration_date, scope, impl, no_register, register_after_upload, summary, guid, protocol, pfn, lfn, transfer_timeout, recursive):
+def upload_command(ctx, file_paths, rse, lifetime, expiration_date, scope, impl, no_register, register_after_upload, summary, guid, protocol, pfn, lfn, transfer_timeout, recursive, new_file: bool):
     """Upload file(s) to a Rucio RSE"""
     args = Arguments(
```

Comment:
2. Another safety mechansim: one could consider an Interactive confirmation, e.g.

```suggestion
    # Safety check for --new-file flag
    if new_file and not ctx.obj.no_pager:  # Interactive session check
        click.confirm(
            'WARNING: --new-file skips existence and checksum validation. '
            'Using this on already-registered files will cause silent data corruption. '
            'Are you sure you want to continue?',
            abort=True
        )
    
    args = Arguments(
```

Better then 1.